### PR TITLE
[MS] Add workspace contextual menu in sidebar

### LINKED ISSUE #6966: Add option menu (popover) on workspaces in Sidebar
Image

### PR DESCRIPTION
**Pull request by Ironicbay**
Closes #6966 

## Checklist

Before you submit this pull request, please make sure to:

- [x] Keep changes in the pull request as small as possible
   
- [x] Ensure the commit history is sanitized
   
- [x] Give a meaningful title to your PR
- [x] Describe your changes
   
- [x] Link any related issue in the description
   
- [x] Link any dependent pull request in the description

If this PR adds a new feature or fixes a bug:

- [x] I have added or updated relevant tests

If this PR is related to GUI:

- [x] I have checked that GUIs affected by my changes are working properly.

### CODE REVIEW

File: client/src/components/workspaces/utils.ts [hunk 1]
```diff
@@ -46,3 +63,164 @@ export function canChangeRole(
...
+      clientRole: workspace.currentSelfRole,
+    },
+  });
+  await popover.present();
+
+  const { data } = await popover.onDidDismiss();
+  if (data !== undefined) {
+    switch (data.action) {
+      case WorkspaceAction.Share:
+        return workspaceShareClick(new Event('ignored'), workspace, informationManager);
```
Comment (Max-7) [inline client/src/components/workspaces/utils.ts:110]: Missing an await.

File: client/src/components/workspaces/WorkspaceCard.vue [hunk 2]
```diff
@@ -3,11 +3,11 @@
 <template>
   <div
     class="card"
-    @click="$emit('click', $event, workspace)"
+    @click="$emit('click', workspace)"
```
Comment (Max-7) [inline client/src/components/workspaces/WorkspaceCard.vue:6]: The `$event` should still be present here.

Don't see the use of this component only in this particular context. This component should be usable in different ways. What if we want to open a context menu on click? We need the event to know where to position it.

All the emits should take an optional `Event` parameter, and if we can provide it, we do.

This component provides an API, it shouldn't be strongly coupled with the use we make of it. Instead, it should be as generic as possible.

This comment stands for other similar emits.

File: client/src/components/workspaces/utils.ts [hunk 3]
```diff
@@ -46,3 +63,161 @@ export function canChangeRole(
...
+        await copyLinkToClipboard(workspace, informationManager);
+        break;
+      case WorkspaceAction.OpenInExplorer:
+        await openWorkspace(workspace, informationManager);
+        break;
+      case WorkspaceAction.Rename:
+        await openRenameWorkspaceModal(workspace, informationManager);
+        break;
+      default:
+        break;
```
Comment (Max-7) [inline client/src/components/workspaces/utils.ts:118]: Add a little warning log, we should never reach the default.

**Review by Max-7 — CHANGES_REQUESTED** [review]

**Review by Max-7 — COMMENTED** [review]
Also would benefit from a few tests. No need to go too deep, but at least check that the button appears and opens the context menu and the context menu contains the expected actions.

**Review by NicoTuxx — APPROVED** [review]

**Review by Max-7 — APPROVED** [review]

**Pull request merged**